india_compliance/gst_india/utils/e_invoice.py

```diff
@@ -127,6 +127,18 @@ def generate_e_invoice(docname, throw=True, force=False):
         # Handle Duplicate IRN
         if result.InfCd == "DUPIRN":
             response = api.get_e_invoice_by_irn(result.Desc.Irn)
+            if signed_data := response.SignedInvoice:
+                invoice_data = json.loads(
+                    jwt.decode(signed_data, options={"verify_signature": False})["data"]
+                )
+
+                if invoice_data.get("ValDtls").get("TotInvVal") != doc.grand_total:
+                    frappe.throw(
+                        _(
+                            "You seem to have a valid e-invoice against this Invoice number with different details."
+                            " You should generate a new e-Invoice with a different Invoice Number if the details are different."
```

Comment:
Sounds a little vague. Give references. You can also give grand total from the comparison made above.
You can start with an e-Invoice was already available against Invoice SINV-* with a Grand Total of Rs 1000.